packages/net-packages/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientAssemblyNamespaceAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.ClientSdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ClientAssemblyNamespaceAnalyzer : DiagnosticAnalyzer
+    {
+        internal static readonly string[] AllowedNamespacePrefix = new[]
+        {
+            "Azure.Diagnostics",
+            "Azure.Cognitive",
+            "Azure.Iot",
+            "Azure.Networking",
+            "Azure.Runtime",
+            "Azure.Security",
+            "Azure.Storage"
```

Comment:
cc @maririos 

I took this list from https://github.com/Azure/azure-sdk/blob/master/docs/design/dotnet/DesignGuidelines.mdk#L225 and `Azure.ApplicationModel` is not in it. Is it registered? Should we add it to the guide?

Reply:
Should we make this a configuration file instead of hardcoding it?

Reply:
I think this would change very seldomly and is not worth the added complexity.

Reply:
Ohh interesting. I do know that there has been some debate with the name .NET chose.
do you know who approves it? 
FYI @KrzysztofCwalina